components/server/src/projects/projects-service.spec.db.ts

```diff
@@ -0,0 +1,208 @@
+import { DBTeam } from "@gitpod/gitpod-db/lib/typeorm/entity/db-team";
+import { DBProject } from "@gitpod/gitpod-db/lib/typeorm/entity/db-project";
+import { Experiments } from "@gitpod/gitpod-protocol/lib/experiments/configcat-server";
+import * as chai from "chai";
+import { Container, ContainerModule } from "inversify";
+import "mocha";
+import { v4 as uuidv4 } from "uuid";
+import { Authorizer } from "../authorization/authorizer";
+import { SpiceDBClient } from "../authorization/spicedb";
+import { SpiceDBAuthorizer } from "../authorization/spicedb-authorizer";
+import { OrganizationService } from "../orgs/organization-service";
+import { ProjectsService } from "./projects-service";
+import { Config } from "../config";
+import { AuthProviderService } from "../auth/auth-provider-service";
+import { IAnalyticsWriter, NullAnalyticsWriter } from "@gitpod/gitpod-protocol/lib/analytics";
+import { HostContainerMapping } from "../auth/host-container-mapping";
+import { HostContextProvider, HostContextProviderFactory } from "../auth/host-context-provider";
+import { AuthProviderParams } from "../auth/auth-provider";
+import { HostContextProviderImpl } from "../auth/host-context-provider-impl";
+import { Organization, User } from "@gitpod/gitpod-protocol";
+import { ResponseError } from "vscode-ws-jsonrpc";
+
+const expect = chai.expect;
+
+describe("ProjectsService", async () => {
+    let container: Container;
+    let owner: User;
+    let stranger: User;
+    let org: Organization;
+
+    beforeEach(async () => {
+        container = testContainer.createChild();
+        container.load(
+            new ContainerModule((bind) => {
+                bind(OrganizationService).toSelf().inSingletonScope();
+                bind(ProjectsService).toSelf().inSingletonScope();
+                bind(Config).toConstantValue({});
+                bind(AuthProviderService).toSelf().inSingletonScope();
+                bind(IAnalyticsWriter).toConstantValue(NullAnalyticsWriter);
+                // hostcontext
+                bind(HostContainerMapping).toSelf().inSingletonScope();
+                bind(HostContextProviderFactory)
+                    .toDynamicValue(({ container }) => ({
+                        createHostContext: (config: AuthProviderParams) =>
+                            HostContextProviderImpl.createHostContext(container, config),
+                    }))
+                    .inSingletonScope();
+                bind(HostContextProvider).to(HostContextProviderImpl).inSingletonScope();
+
+                // auth
+                bind(SpiceDBClient)
+                    .toDynamicValue(() => {
+                        const token = uuidv4();
+                        return v1.NewClient(token, "localhost:50051", v1.ClientSecurity.INSECURE_PLAINTEXT_CREDENTIALS)
+                            .promises;
+                    })
+                    .inSingletonScope();
```

Comment:
Can we/does it make sense, to extract this constructor and re-use between the org/project services?

Reply:
did that in the other PR already